lib/lf/mesh/hybrid2d/mesh_factory.h

```diff
@@ -51,6 +51,12 @@ class MeshFactory : public mesh::MeshFactory {
   // NOLINTNEXTLINE(modernize-use-nodiscard)
   size_type AddPoint(std::unique_ptr<geometry::Geometry>&& geometry) override;
 
+  [[nodiscard]] geometry::Point const& GetPoint(size_type index) const noexcept {
+    assert(index < nodes_.size());
+    assert(dynamic_cast<geometry::Point const*>(nodes_[index].get()));
+    return static_cast<geometry::Point const&>(*nodes_[index]);
```

Comment:
Please use `dynamic_cast`, this way we are safe in case `geometry::Point` should start using virtual inheritance at some point.